test/e2e/volume_provisioning.go

```diff
@@ -304,12 +303,12 @@ var _ = framework.KubeDescribe("Dynamic provisioning", func() {
 
 			By("creating an external dynamic provisioner pod")
 			pod := startExternalProvisioner(c, ns)
-			defer c.Core().Pods(ns).Delete(pod.Name, nil)
+			defer framework.DeletePodOrFail(c, ns, pod.Name)
```

Comment:
Would be nice to create a separate pr that allows for `apierrs.IsNotFound` in `DeletePodOrFail`.